src/cpp/src/visual_language/embedding_model.cpp

```diff
@@ -71,7 +71,9 @@ ov::Tensor EmbeddingsModel::infer(const ov::Tensor& input_idx, bool return_remot
         m_request.set_output_tensor(m_cpu_tensor);
     }
     m_request.infer();
-    return m_request.get_output_tensor();
+    ov::Tensor out = m_request.get_output_tensor();
+    OPENVINO_ASSERT(input_idx.get_shape().at(1) == out.get_shape().at(1));
```

Comment:
why do we need it?

Reply:
That's the requested assert to ensure that tokenized representation has the same length as input embeddings in a common place.